pytube/__main__.py

```diff
@@ -91,6 +91,11 @@ def __init__(
     def __repr__(self):
         return f'<pytube.__main__.YouTube object: videoId={self.video_id}>'
 
+    def __eq__(self, o: object) -> bool:
+        # Compare types and urls, if they're same return true, else return false.
+        if type(o) == type(self) and o.watch_url == self.watch_url: return True
+        else: return False
```

Comment:
Why not use the following instead?

`return type(o) == type(self) and o.watch_url == self.watch_url`

Reply:
oh yeah, sorry i am dumb